src/components/Transactions/Transactions.js

```diff
@@ -56,23 +61,42 @@ const TransactionsSkeleton = () => (
   </Card>
 );
 
-const Transactions = ({
-  transactions,
-  toggleTransactionsList,
-  isTransactionsItemsLoading,
-  isExpanded,
-  isShowMoreButtonHidden,
-  error,
-  isTransactionsSectionLoading,
-  t
-}) =>
-  isTransactionsSectionLoading ? (
+const Transactions = ({ t }) => {
+  const {
+    transactionsList,
+    showToggleButton,
+    transactionsError,
+    isTransactionsSectionLoading,
+    isTransactionListExpanded
+  } = useSelector(state => state.transactions);
+
+  const transactionsToShow = isTransactionListExpanded
+    ? transactionsList
+    : transactionsList.slice(0, DEFAULT_TRANSACTIONS_NUMBER);
+
+  const dispatch = useDispatch();
+
+  const toggleTransactionsList = () => {
+    if (isTransactionListExpanded) {
+      dispatch(setIsTransactionListExpanded(false));
+    } else {
+      dispatch(setIsTransactionListExpanded(true));
+    }
+  };
+
+  useEffect(() => {
+    if (transactionsList?.length === 0) {
+      dispatch(fetchListCustomerTransactions());
+    }
+  }, []);
+
```

Comment:
if(isTransactionsSectionLoading)
  return <TransactionsSkeleton />

return (... ) //regular component

With this solution, we will skip the nesting